vowpalwabbit/reduction_stack.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include "reductions_fwd.h"
+
+struct vw;
+
+typedef VW::LEARNER::base_learner* (*reduction_setup_fn)(VW::setup_base_fn&, VW::config::options_i&, vw&);
+
+namespace VW
+{
+struct status_quo : public setup_base_fn
+{
+  std::vector<std::string> enabled_reductions;
+
+  void print_enabled_reductions(vw& all);
+
+  status_quo(vw& all);
+
+  // this function consumes all the reduction_stack until it's able to construct a base_learner
+  // same signature as the old setup_base(...) from parse_args.cc
```

Comment:
nit: `same signature as ...` do we want this as a comment in the code? maybe a comment in the PR for tracking?